### ISSUE TITLE
Pass starting Design Type into ODOP

### ISSUE THREAD

**Issue by grumpyinca** [thegrumpys/odop#388]
The SpringDesignSoftware.org web site is designed to have stand-alone landing pages that are specific to compression, extension & torsion springs.  These pages can provide a URL that encodes a preferred Design Type and Startup design (or not).  Once the Okta login is deferred (See #364), it should be possible for users with a design type preference to be taken directly to a startup of the design type and units (US vs. metric) of their choice. Users coming from the website getting-started page can be served by the original promptForDesign or the File : Open modal.

**Comment by 1fifoto** [comment]
Sample URLS:

URL|Display PromptForDesign
---|---
|http://localhost:3000 | Yes
http://localhost:3000/designtype/Spring%2FCompression/design/Startup | No
http://localhost:3000/designtype/Spring%2FExtension/design/Startup | No
http://localhost:3000/designtype/Spring%2FTorsion/design/Startup | No

**Comment by 1fifoto** [comment]
Problem: If the user needs to login then the parameters are somehow deleted from the URL.

**Comment by 1fifoto** [comment]
The preferred URL approach is to use query parameters and not URL parameters, such as, http://localhost:3000/?designtype=Spring%2FCompression&design=Startup. This then allows all internal URLs to still be relative and not forcing them to be absolute, for example, src="spinner.gif" instead of src="/sprinner.gif".

**Comment by 1fifoto** [comment]
Also consider adding execute=name, demo=name and tutor=name as optional query parameters

**Comment by 1fifoto** [comment]
Sample URLS using query parameters (preferred):

URL|type|name|Display PromptForDesign
---|---|---|---
http://localhost:3000 | REACT_APP_DESIGN_TYPE | REACT_APP_DESIGN_NAME | Yes
http://localhost:3000/?name=Startup |  REACT_APP_DESIGN_TYPE | Startup | No
http://localhost:3000/?type=Spring%2FCompression |  Spring/Compression | REACT_APP_DESIGN_NAME | No
http://localhost:3000/?type=Spring%2FCompression&name=Startup |  Spring/Compression | Startup | No
http://localhost:3000/?type=Spring%2FExtension |  Spring/Extension | REACT_APP_DESIGN_NAME | No
http://localhost:3000/?type=Spring%2FExtension&name=Startup |  Spring/Extension | Startup | No
http://localhost:3000/?type=Spring%2FTorsion |  Spring/Torsion | REACT_APP_DESIGN_NAME | No
http://localhost:3000/?type=Spring%2FTorsion&name=Startup |  Spring/Torsion | Startup | No

**Comment by 1fifoto** [comment]
Added "view" parameter.

Sample URL for "Report1"

    http://localhost:3000/?type=Spring%2FCompression&name=Startup&view=Report1

Sample screen shot:
![Screen Shot 2020-12-02 at 8 01 56 PM](https://user-images.githubusercontent.com/1581602/100954006-5abfd100-34d9-11eb-9412-d46eb3fbdd7b.png)

**Comment by 1fifoto** [comment]
Waiting on #438 before continuing on this issue.

**Comment by 1fifoto** [comment]
I have done type=,name= and view= and checked them into branch 388. I still need to do exec=.

**Comment by 1fifoto** [comment]
Finally got it working. The parameter is "execute=".

http://localhost:3000/?type=Spring%2FExtension&name=StartUp&view=View1&execute=demo

**Comment by 1fifoto** [comment]
Sample links to use for testing execute=

URL|type|name|execute|Result
---|---|---|---|---
https://odop-staging.herokuapp.com/?execute=demo | REACT_APP_DESIGN_TYPE = Compression | REACT_APP_DESIGN_NAME = StartUp | demo | Success
https://odop-staging.herokuapp.com/?name=Startup&execute=demo |  REACT_APP_DESIGN_TYPE = Compression | Startup | demo | Success
https://odop-staging.herokuapp.com/?type=Spring%2FCompression |  Spring/Compression | REACT_APP_DESIGN_NAME = StartUp | demo | Fail
https://odop-staging.herokuapp.com/?type=Spring%2FCompression&name=Startup |  Spring/Compression | Startup | demo | Fail
https://odop-staging.herokuapp.com/?type=Spring%2FExtension |  Spring/Extension | REACT_APP_DESIGN_NAME = StartUp | demo6 | Fail
https://odop-staging.herokuapp.com/?type=Spring%2FExtension&name=Startup |  Spring/Extension | Startup | demo6 | Fail
https://odop-staging.herokuapp.com/?type=Spring%2FTorsion |  Spring/Torsion | REACT_APP_DESIGN_NAME = StartUp | demo14 | Fail
https://odop-staging.herokuapp.com/?type=Spring%2FTorsion&name=Startup |  Spring/Torsion | Startup | demo14 | Fail

**Comment by 1fifoto** [comment]
Same as above except for odop-staging.herokuapp.com

URL|type|name|execute|Result
---|---|---|---|---
https://odop-staging.herokuapp.com/?execute=demo | REACT_APP_DESIGN_TYPE = Compression | REACT_APP_DESIGN_NAME = StartUp | demo | Success
https://odop-staging.herokuapp.com/?name=Startup&execute=demo |  REACT_APP_DESIGN_TYPE = Compression | Startup | demo | Success
https://odop-staging.herokuapp.com/?type=Spring%2FCompression&execute=demo |  Spring/Compression | REACT_APP_DESIGN_NAME = StartUp | demo | Success
https://odop-staging.herokuapp.com/?type=Spring%2FCompression&name=Startup&execute=demo |  Spring/Compression | Startup | demo | Success
https://odop-staging.herokuapp.com/?type=Spring%2FExtension&execute=demo6 |  Spring/Extension | REACT_APP_DESIGN_NAME = StartUp | demo6 | Success
https://odop-staging.herokuapp.com/?type=Spring%2FExtension&name=Startup&execute=demo6 |  Spring/Extension | Startup | demo6 | Success
https://odop-staging.herokuapp.com/?type=Spring%2FTorsion&execute=demo14 |  Spring/Torsion | REACT_APP_DESIGN_NAME = StartUp | demo14 | Success
https://odop-staging.herokuapp.com/?type=Spring%2FTorsion&name=Startup&execute=demo14 |  Spring/Torsion | Startup | demo14 | Success

**Comment by grumpyinca** [comment]
Latest (12/30) round of testing.

Background:
Branches 
 - 455 (Extension, torsion Calculator constraint highlighting)
 - 459 (Develop "Welcome" execute sessions) 
 - 463 (Improve FIX header and tooltip)  

were coded and merged to master.

Master was merged to branch 388.

1. 
Launch with http://localhost:3000/
Observe (as expected) compression spring Startup 
Sign in as ODOP_Tester
File : Open, drop down "Select private or system design to open:" selection list
Observe private designs named bbb, fff, sss associated with Bri's account
Don't see expected private designs associated with ODOP_Tester account.

Same problem appears on the staging system (I really don't remember what private designs are in that staging account).

2a. 
Start signed out (shouldn't matter)
File : Open HotWoundMetric  (should clear AutoSave data)
Close Browser tab.
Re-launch with: http://localhost:3000/?type=Spring%2FCompression&name=Startup&view=View1&execute=welcome1
See popup "Migrated design from version 8 to version 9". (seems correct for Startup)
Why is the design named InitialState ?

2b. 
Try to switch Calculator --> Advanced view. It doesn't work.
Switch to Report1 (or any other) View.  It works.
Switch to Advanced view.  Now it works.  Huh ?

2b also happens with:
https://odop-staging.herokuapp.com/?type=Spring%2FCompression&name=Startup&view=View1&execute=demo

**Comment by 1fifoto** [comment]
# 1 - Fixed with code change and checked-in. The SignIn code was using Application Id instead of User Id.
# 2a - Your "welcome1" script has ''loadInitialState('Spring/Compression')," in it that is causing the fetched model named "StartUp" to be overwritten with the one named "initialState". Delete the line. I did not check in any changed for this.
# 2b - Fixed with code change and checked-in. Needed to set the view (changeBiew(view)) into the Redux model in MainPage constructor.

**Comment by 1fifoto** [comment]
Merged into master, closing

**Comment by grumpyinca** [comment]
Refer to #477 for detail on startup with the File : Open modal.

**Comment by grumpyinca** [comment]
See: [URL syntax](https://thegrumpys.github.io/odop/design/URLsyntax)
